include/ebpf_program_types.h

```diff
@@ -36,6 +36,7 @@ typedef struct _ebpf_helper_function_prototype
     ebpf_return_type_t return_type;
     ebpf_argument_type_t arguments[5];
     ebpf_helper_function_prototype_flags_t flags;
+    bool implicit_context;
```

Comment:
Note: `implicit_context` is added as a separate bool instead of a bit field in `ebpf_helper_function_prototype_flags_t` to handle forward compatibility issues.

The `implicit_context` change is optional and backward compatible but is not forward compatible. This means that is an extension is compiled with an older (<= 0.17.x) version of eBPF headers and is deployed on a machine running a newer (>= 0.18.x) version of eBPF runtime, it can safely run with no issues.

On the other hand, if the extension is compiled with a newer version of eBPF and supports implicit context, and gets deployed with older eBPF runtime, it can cause a crash unless eBPF runtime rejects binding with such an extension.

Adding a new `implicit_context`  in `ebpf_helper_function_prototype_t` changes the size of the struct allowing eBPF to detect and reject incmpatible version of extension.

Reply:
I agree with blocking this, but I am not sure overloading the verification check as being the correct approach.

Reply:
Re-opening as I have a question about this - In the future, when would it be appropriate to add to the flags field vs requiring a new bool?

Reply:
So I think if the new feature is both backward and forward compatible, it can be added to the existing flags. Agree that we can probably document this somewhere, but not sure which doc (ebpfExtension.md is ideally for extension developers, but this will be for eBPF developers). I can maybe add a comment in this file itself?

Reply:
Updated doc to mention this point.